fem/fe.cpp

```diff
@@ -1750,6 +1750,235 @@ void BiQuad2DFiniteElement::ProjectDelta(int vertex, Vector &dofs) const
 #endif
 }
 
+
+H1Ser_QuadrilateralElement::H1Ser_QuadrilateralElement(const int p)
+   : ScalarFiniteElement(2, Geometry::SQUARE, (p*p + 3*p +6) / 2, p,
+                         FunctionSpace::Qk)
+{
+   // Store the dof_map of the associated TensorBasisElement, which will be used
+   // to create the serendipity dof map.  Its size is larger than the size of
+   // the serendipity element.
+   TensorBasisElement *tbeTemp =
```

Comment:
Why use `new`?

Reply:
I'm still pretty new to C++... I guess you are saying that I could do this instead:
`TensorBasisElement tbeTemp =
      TensorBasisElement(...)'`
`const Array<int> tp_dof_map = tbeTemp.GetDofMap();`
Is that preferable for some reason?  Happy to make the change - just trying to learn.

Reply:
It is generally a good practice to avoid using pointers when it is unnecessary. Pointers are good sources of bugs. By creating an object rather than a pointer, the desctructor will be called automatically when the object goes out of scope. With a pointer, one must remember to delete at the end of the scope, which is easy in simple cases like this but can be tricky sometimes.